### PR TITLE
fix cf route53 deployment, add tests

### PR DESCRIPTION
**Pull request by baermat**
fix discrepancy between the cf route53 model and the called operation

### CODE REVIEW

File: localstack/services/cloudformation/models/route53.py [hunk 1]
```diff
@@ -40,16 +40,36 @@ def param_change_batch(params, **kwargs):
...
+        def hosted_zone_id_change_batch(params, **kwargs):
+            route53 = aws_stack.connect_to_service("route53")
+            hosted_zone_id = params.get("HostedZoneId")
+            if not hosted_zone_id:
+                hosted_zone_name = params.get("HostedZoneName")
+                # https://docs.aws.amazon.com/Route53/latest/APIReference/API_ChangeResourceRecordSets.html"
+                # "Specify either HostedZoneName or HostedZoneId, but not both. If you have multiple hosted zones with
+                # the same domain name, you must specify the hosted zone using HostedZoneId."
+                if not hosted_zone_name:
+                    raise Exception("Either HostedZoneId or HostedZoneName must be present.")
```
Comment (baermat) [inline localstack/services/cloudformation/models/route53.py:58]: Not happy about throwing Exception here, but also unsure which one I should pick here

File: tests/integration/cloudformation/test_cloudformation_route53.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    cleanup_stacks,
+):
+    stack_name = f"stack-{short_uid()}"
+    change_set_name = f"change-set-{short_uid()}"
+    route53_name = f"www.{short_uid()}.com."
+    caller_ref = f"caller-ref-{short_uid()}"
+    route53_client.create_hosted_zone(Name=route53_name, CallerReference=caller_ref)
+
+    template_rendered = load_template_raw("route53_hostedzonename_template.json")
+    template_rendered = template_rendered % (route53_name, route53_name)
```
Comment (dominikschubert) [inline tests/integration/cloudformation/test_cloudformation_route53.py:61]: Please use the jinja template mapping for this (see `deploy_cfn_template` fixture). It's a bit hard to maintain with this positional formatting. Alternatively you could use CloudFormation Parameters to specify these names as well.

File: tests/integration/cloudformation/test_cloudformation_route53.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    is_change_set_created_and_available,
+    cleanup_changesets,
+    cleanup_stacks,
+    is_stack_created,
+):
+    stack_name = f"stack-{short_uid()}"
+    change_set_name = f"change-set-{short_uid()}"
+    route53_name = f"www.{short_uid()}.com."
+    caller_ref = f"caller-ref-{short_uid()}"
+    create_zone_response = route53_client.create_hosted_zone(
```
Comment (dominikschubert) [inline tests/integration/cloudformation/test_cloudformation_route53.py:18]: missing resource cleanup here

File: tests/integration/cloudformation/test_cloudformation_route53.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    try:
+        wait_until(is_change_set_created_and_available(change_set_id))
+        cfn_client.execute_change_set(ChangeSetName=change_set_id)
+        wait_until(is_stack_created(stack_id))
+        describe_result = cfn_client.describe_stacks(StackName=stack_id)
+        assert describe_result["Stacks"][0]["StackStatus"] == "CREATE_COMPLETE"
+
+    finally:
+        cleanup_changesets([change_set_id])
+        cleanup_stacks([stack_id])
```
Comment (dominikschubert) [inline tests/integration/cloudformation/test_cloudformation_route53.py:43]:
```suggestion
    deploy_cfn_template(template_filename="route53_hostedzoneid_template.json")
```

could be replaced with the usage of the cfn fixture

File: tests/integration/cloudformation/test_cloudformation_route53.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        StackName=stack_name,
+        ChangeSetName=change_set_name,
+        TemplateBody=template_rendered,
+        ChangeSetType="CREATE",
+    )
+
+    change_set_id = response["Id"]
+    stack_id = response["StackId"]
+    try:
+        wait_until(is_change_set_created_and_available(change_set_id))
```
Comment (dominikschubert) [inline tests/integration/cloudformation/test_cloudformation_route53.py:72]: same as above. can be replaced with `deploy_cfn_template`

File: tests/integration/templates/route53_hostedzoneid_template.json [hunk 6]
```diff
@@ -0,0 +1,14 @@
+{
+"Resources": {
+  "myDNSWithIdRecord": {
+    "Type": "AWS::Route53::RecordSet",
+    "Properties": {
+      "HostedZoneId": "%s",
+      "Name": "%s",
```
Comment (dominikschubert) [inline tests/integration/templates/route53_hostedzoneid_template.json:7]: would suggest replacing these with named jinja parameters or cloudformation parameters

File: tests/integration/templates/route53_hostedzonename_template.json [hunk 7]
```diff
@@ -0,0 +1,14 @@
+{
+"Resources": {
+  "myDNSWithNameRecord": {
+    "Type": "AWS::Route53::RecordSet",
+    "Properties": {
+      "HostedZoneName": "%s",
+      "Name": "%s",
```
Comment (dominikschubert) [inline tests/integration/templates/route53_hostedzonename_template.json:7]: same was comment above

File: localstack/services/cloudformation/models/route53.py (hunk 1 above)
Comment (dominikschubert) [reply localstack/services/cloudformation/models/route53.py:58]: Do you mean the exception class or the wording of the message?

File: localstack/services/cloudformation/models/route53.py [hunk 8]
```diff
@@ -40,16 +40,36 @@ def param_change_batch(params, **kwargs):
...
+                hosted_zones = route53.list_hosted_zones_by_name(DNSName=hosted_zone_name)[
+                    "HostedZones"
+                ]
+                if not len(hosted_zones) == 1:
```
Comment (dominikschubert) [inline localstack/services/cloudformation/models/route53.py:62]:
```suggestion
                if len(hosted_zones) != 1:
```

**Review by dominikschubert — CHANGES_REQUESTED** [review]
Thanks for tackling this :muscle: 

Left some comments that need to be addressed so that the code conforms to our most up to date tooling :+1:

File: localstack/services/cloudformation/models/route53.py (hunk 1 above)
Comment (baermat) [reply localstack/services/cloudformation/models/route53.py:58]: Exception class

File: tests/integration/cloudformation/test_cloudformation_route53.py (hunk 3 above)
Comment (baermat) [reply tests/integration/cloudformation/test_cloudformation_route53.py:18]: thanks! good catch

File: tests/integration/cloudformation/test_cloudformation_route53.py (hunk 4 above)
Comment (baermat) [reply tests/integration/cloudformation/test_cloudformation_route53.py:43]: alright, I think I simply took a look on some existing test to copy the behaviour, but will do ofc :+1: .

File: tests/integration/templates/route53_hostedzonename_template.json (hunk 7 above)
Comment (baermat) [reply tests/integration/templates/route53_hostedzonename_template.json:7]: :+1:

File: tests/integration/templates/route53_hostedzoneid_template.json (hunk 6 above)
Comment (baermat) [reply tests/integration/templates/route53_hostedzoneid_template.json:7]: :+1:

File: tests/integration/cloudformation/test_cloudformation_route53.py (hunk 5 above)
Comment (baermat) [reply tests/integration/cloudformation/test_cloudformation_route53.py:72]: :+1:

File: tests/integration/cloudformation/test_cloudformation_route53.py (hunk 2 above)
Comment (baermat) [reply tests/integration/cloudformation/test_cloudformation_route53.py:61]: :+1:

File: localstack/services/cloudformation/models/route53.py (hunk 8 above)
Comment (baermat) [reply localstack/services/cloudformation/models/route53.py:62]: :+1:

File: localstack/services/cloudformation/models/route53.py (hunk 1 above)
Comment (dominikschubert) [reply localstack/services/cloudformation/models/route53.py:58]: Don't think we have anything special here right now to indicate these, so I'd say it's fine as is :+1:

**Review by dominikschubert — APPROVED** [review]
LGTM :+1:

**Pull request merged**